framework/web/Response.php

```diff
@@ -488,6 +485,9 @@ public function sendContentAsFile($content, $attachmentName, $options = [])
             $this->content = $content;
         }
 
+        $mimeType = isset($options['mimeType']) ? $options['mimeType'] : 'application/octet-stream';
+        $this->setDownloadHeaders($attachmentName, $mimeType, !empty($options['inline']), $end - $begin + 1);
```

Comment:
Should use `StringHelper::byteLength($this->content)` as `$begin` and `$end` are obtained from request header and may not be correct.